src/ui/public/vis/request_handlers/courier.js

```diff
@@ -137,10 +136,10 @@ const CourierRequestHandlerProvider = function (Private, courier, timefilter) {
               searchSource.rawResponse = resp;
               return _.cloneDeep(resp);
             }).then(async resp => {
-              for (const agg of vis.getAggConfig()) {
+              for (const agg of aggs) {
                 if (_.has(agg, 'type.postFlightRequest')) {
-                  const nestedSearchSource = new SearchSource().inherits(requestSearchSource);
-                  resp = await agg.type.postFlightRequest(resp, vis.aggs, agg, nestedSearchSource);
+                  const nestedSearchSource = requestSearchSource.makeChild();
```

Comment:
Just to note that down: this will change the behavior slightly, since we now call all parents start handler (i.e. the one of `requestSearchSource` in case this nestedSearchSource will receive the `requestIsStarting`. I am not yet sure if that actually will ever happen of what that influences, but wanted to point it out for now.